lib/fabrik/CfPlatformManager.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const Promise = require('bluebird');
+const BasePlatformManager = require('./BasePlatformManager');
+const utils = require('../utils');
+const assert = require('assert');
+const errors = require('../errors');
+const cloudController = require('../cf').cloudController;
+const logger = require('../logger');
+const SecurityGroupNotCreated = errors.SecurityGroupNotCreated;
+const SecurityGroupNotFound = errors.SecurityGroupNotFound;
+const NotImplemented = errors.NotImplemented;
+const ordinals = ['First', 'Second', 'Third', 'Fourth', 'Fifth', 'Sixth', 'Seventh', 'Eighth', 'Ninth', 'Tenth'];
+
+class CfPlatformManager extends BasePlatformManager {
+  constructor(guid, manager, context) {
+    super(guid, manager, context);
+    this.space_guid = context.space_guid;
+    this.cloudController = cloudController;
+  }
+
+  preInstanceProvisionOperations(options) {
+    /* jshint unused:false */
+    throw new NotImplemented('preInstanceProvisionOperations');
+  }
+
+  postInstanceProvisionOperations(options) {
+    return this.createSecurityGroup(this.space_guid, options.ruleId, options.ipRules);
+  }
+
+  preInstanceDeleteOperations(options) {
+    return this.deleteSecurityGroup(options.ruleId);
+  }
+
+  postInstanceDeleteOperations(options) {
+    /* jshint unused:false */
+    throw new NotImplemented('postInstanceDeleteOperations');
+  }
+
+  createSecurityGroup(space_guid, name, rules) {
+    logger.info(`Creating security group '${name}' with rules ...`, rules);
+    return utils
```

Comment:
No point in having this class if still DockerInstance and DirectorInstance classes having code constructs which are CF specific. One can rename the method to `buildIpRules` but its still CF specific constructs that are being returned back & they will break for K8s. Its better to avoid this class all together if it cant be refactored the right way.

Reply:
'buildIpRules' --> should be refactored and it should only send the params to build the rule (in cf) or network policy (in K8S). Want to avoid further segregation of docker and director related code platform manager. Yes, agree the buildIpRules method now send the rule (fits only for K8S) but will send ips and  ports in future. As part of dynamic ip feature we will take that up (I remember we have decided this.)

Reply:
This still does not seem right :-) lemme try something hands on later.